test/e2e/delete_test.go

```diff
@@ -22,59 +20,49 @@ func deletionTestSuite(t *testing.T) {
 
 // testWindowsNodeDeletion tests the Windows node deletion from the cluster.
 func testWindowsNodeDeletion(t *testing.T) {
-	testCtx, err := NewTestContext(t)
+	testCtx, err := NewTestContext()
 	require.NoError(t, err)
 
 	// Get all the Machines in the
-	machineSetList := &mapi.MachineSetList{}
-	listOpts := []client.ListOption{
-		client.InNamespace("openshift-machine-api"),
-	}
-	err = framework.Global.Client.List(context.TODO(), machineSetList, listOpts...)
-	require.NoError(t, err)
+	machineSetList, err := testCtx.client.Machine.MachineSets(clusterinfo.MachineAPINamespace).List(context.TODO(), meta.ListOptions{})
+	require.NoError(t, err, "error listing MachineSets")
 	var windowsMachineSetWithLabel string
-	var e2eMachineSetNames []string
+	var e2eMachineSets []*mapi.MachineSet
 	for _, machineSet := range machineSetList.Items {
 		if strings.Contains(machineSet.Name, "e2e-windows-machineset-") {
-			e2eMachineSetNames = append(e2eMachineSetNames, machineSet.Name)
+			e2eMachineSets = append(e2eMachineSets, &machineSet)
 		}
 		if machineSet.Spec.Selector.MatchLabels["machine.openshift.io/os-id"] == "Windows" {
 			windowsMachineSetWithLabel = machineSet.Name
 		}
 	}
-	windowsMachineSet := &mapi.MachineSet{}
-	err = framework.Global.Client.Get(context.TODO(), types.NamespacedName{Name: windowsMachineSetWithLabel,
-		Namespace: "openshift-machine-api"}, windowsMachineSet)
-	require.NoError(t, err)
+	windowsMachineSet, err := testCtx.client.Machine.MachineSets(clusterinfo.MachineAPINamespace).Get(context.TODO(),
+		windowsMachineSetWithLabel, meta.GetOptions{})
+	require.NoError(t, err, "error getting Windows MachineSet")
 	// Reset the number of nodes to be deleted to 0
 	gc.numberOfNodes = 0
 	// Delete the Windows VM that got created.
 	windowsMachineSet.Spec.Replicas = &gc.numberOfNodes
-	if err := framework.Global.Client.Update(context.TODO(), windowsMachineSet); err != nil {
-		t.Fatalf("error updating windowsMachineSet custom resource  %v", err)
-	}
+	_, err = testCtx.client.Machine.MachineSets(clusterinfo.MachineAPINamespace).Update(context.TODO(), windowsMachineSet,
+		meta.UpdateOptions{})
+	require.NoError(t, err, "error updating Windows MachineSet")
 	// we are waiting 10 minutes for all windows machines to get deleted.
 	err = testCtx.waitForWindowsNodes(gc.numberOfNodes, true, true, false)
-	if err != nil {
-		t.Fatalf("windows node deletion failed  with %v", err)
-	}
+	require.NoError(t, err, "Windows node deletion failed")
 
 	// Cleanup all the MachineSets created by us.
-	for _, machineSetName := range e2eMachineSetNames {
-		machineSet := &mapi.MachineSet{}
-		err = framework.Global.Client.Get(context.TODO(), types.NamespacedName{Name: machineSetName,
-			Namespace: "openshift-machine-api"}, machineSet)
-		assert.NoError(t, framework.Global.Client.Delete(context.TODO(), machineSet))
+	for _, machineSet := range e2eMachineSets {
+		assert.NoError(t, testCtx.deleteMachineSet(machineSet), "error deleting MachineSet")
```

Comment:
did we decide to log the error instead of failing here?

Reply:
resolved, we want to log such errors ideally, but it shouldn't be part of this PR.